src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs

```diff
@@ -0,0 +1,424 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DocumentFormat.OpenXml.Features;
+using System;
+
+namespace DocumentFormat.OpenXml.Packaging
+{
+    /// <summary>
+    /// Defines extensions for part relationships
+    /// </summary>
+    public static class OpenXmlPartExtensions
+    {
+        private static TChild InitPart<TChild>(this OpenXmlPartContainer parent, TChild childPart, PartTypeInfo partType, string? relId = null)
+            where TChild : OpenXmlPart
+        {
+            if (parent == null)
+            {
+                throw new ArgumentNullException(paramName: nameof(parent));
+            }
+
+            string contentType = partType.ContentType;
+
+            var partExtension = partType.Extension;
+            parent.Features.GetRequired<IPartExtensionFeature>().Register(contentType, partExtension);
```

Comment:
```suggestion
            parent.Features.Get<IPartExtensionFeature>()?.Register(contentType, partExtension);
```

If it's not available for some reason, no need to track it